elm/wizard.py

```diff
@@ -311,3 +238,310 @@ def chat(self, query,
+    Connecting to the database requires the use of the psycopg2
+    python package, environment variables storing the db user and
+    password, and the specification of other connection paremeters
+    such as host, port, and name. The database has the following
+    columns: id, embedding, chunks, and metadata.
+    """
+
+    def __init__(self, db_host, db_port, db_name,
+                 model=None, token_budget=3500):
+        """
+        Parameters
+        ----------
+        model : str
+            GPT model name, default is the DEFAULT_MODEL global var
+        token_budget : int
+            Number of tokens that can be embedded in the prompt. Note that the
+            default budget for GPT-3.5-Turbo is 4096, but you want to subtract
+            some tokens to account for the response budget.
+        db_host : str
+            Host url for postgres database.
+        db_port : str
+            Port for postres database. ex: '5432'
+        db_name : str
+            Postgres database name.
+        """
+        boto3 = try_import('boto3')
+        psycopg2 = try_import('psycopg2')
+
+        db_user = os.getenv("EWIZ_DB_USER")
+        db_password = os.getenv('EWIZ_DB_PASSWORD')
+        assert db_user is not None, "Must set user for postgres database!"
+        assert db_password is not None, "Must set user for postgres database!"
+
+        self.conn = psycopg2.connect(user=db_user,
+                                     password=db_password,
+                                     host=db_host,
+                                     port=db_port,
+                                     database=db_name)
+
+        self.cursor = self.conn.cursor()
+
+        access_key = os.getenv('AWS_ACCESS_KEY_ID')
+        secret_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+        session_token = os.getenv('AWS_SESSION_TOKEN')
+
+        assert access_key is not None, "Must set AWS_ACCESS_KEY_ID!"
+        assert secret_key is not None, ("Must set AWS_SECRET_ACCESS_KEY!")
+        assert session_token is not None, "Must set AWS_SESSION_TOKEN!"
+        self._aws_client = boto3.client(service_name='bedrock-runtime',
+                                        region_name='us-west-2',
+                                        aws_access_key_id=access_key,
+                                        aws_secret_access_key=secret_key,
+                                        aws_session_token=session_token)
+
+        super().__init__(model, token_budget=token_budget)
+
+    def get_embedding(self, text):
+        """Get the 1D array (list) embedding of a text string
+        as generated by AWS Titan.
+
+        Parameters
+        ----------
+        text : str
+            Text to embed
+
+        Returns
+        -------
+        embedding : list
+            List of float that represents the numerical embedding of the text
+        """
+
+        body = json.dumps({"inputText": text, })
+
+        model_id = self.EMBEDDING_MODEL
```

Comment:
Is this supposed to be the titan model name? Seems like this will still be `'text-embedding-ada-002'` based on the `ApiBase` class. I think you need to set the class variable for this class.

Reply:
@grantbuster When I create the wizard object in the run_app.py code I set the model_id using `EnergyWizardPostgres.EMBEDDING_MODEL = 'amazon.titan-embed-text-v1'` and that seems to do the trick. I was thinking doing it that way would be most consistent with the existing code, would it be better to set the class variable?

Reply:
I think we should set the class variable because in the get embedding method we hard-code the call to the AWS object, right? 

In the docstring we may want to have a description of how this is designed, e.g.:

vector database - postgresql with psycopg2
embedding model - aws titan
llm - GPT4 via Azure deployment